internal/internal_task_handlers.go

```diff
@@ -1694,6 +1694,11 @@ func (i *temporalInvoker) internalHeartBeat(ctx context.Context, details *common
 		}
 	}
 
+	if err != nil {
+		logger := GetActivityLogger(ctx)
```

Comment:
Do we need any protection for this codepath potentially not being an activity context?

Reply:
Not from actual user code. Heartbeat can only be called from activity context.